File: management/vo/audit.go
package vo

// AuditLogVo is the HTTP response shape for a single audit log entry.
type AuditLogVo struct {
	ID           string `json:"id"`
	CreatedAt    string `json:"createdAt"`
	UserID       string `json:"userId"`
	UserName     string `json:"userName"`
	UserEmail    string `json:"userEmail"`
	UserIP       string `json:"userIP"`
	WorkspaceID  string `json:"workspaceId"`
	Action       string `json:"action"`
	Resource     string `json:"resource"`
	ResourceID   string `json:"resourceId"`
	ResourceName string `json:"resourceName"`
	Scope        string `json:"scope"`
	Status       string `json:"status"`
	StatusCode   int    `json:"statusCode"`
	Detail       string `json:"detail,omitempty"`
}

File: fronted/src/api/audit.ts
import request from '@/api/request'

function wsID(): string {
  return localStorage.getItem('active_ws_id') || ''
}

export interface AuditLogVo {
  id: string
  createdAt: string
  userId: string
  userName: string
  userEmail: string
  userIP: string
  workspaceId: string
  action: string
  resource: string
  resourceId: string
  resourceName: string
  scope: string
  status: 'success' | 'failed'
  statusCode: number
  detail?: string
}

export interface AuditListParams {
  action?: string
  resource?: string
  status?: string
  keyword?: string
  from?: string
  to?: string
  page?: number
  pageSize?: number
}

export const listAuditLogs = (params?: AuditListParams) =>
  request.get(`/workspaces/${wsID()}/audit-logs`, params ?? {})
